docs: add "Command Line Tool: REPL" (#214)
GreptimeTeam/docs: docs/user-guide/cli-repl.md

```shell
cargo run -- cli attach --grpc-addr=0.0.0.0:4001 
```

Wait for the REPL's prompt comes up:

```text
Ready for commands. (Hint: try 'help')
> 
```

You can type in the "help" command to see some useful helping messages:

```text
> help

Available commands (case insensitive):
- 'help': print this help
- 'exit' or 'quit': exit the REPL
- 'use <your database name>': switch to another database/schema context
- Other typed in text will be treated as SQL.
  You can enter new line while typing, just remember to end it with ';'.
```

This REPL application has hinting based on your history inputs. You can use "tab" key to complete your inputs when some suggestions pop up. A very useful feature powered by [RustyLine](https://crates.io/crates/rustyline).

Now go on typing in some SQLs to play with GreptimeDB! Here are some examples:

- Create and use databases, you can see the prompt is prefixed with the database you are currently using:

```text
> create database foo;
Affected Rows: 1
Cost 21 ms
> show databases;
+---------+
| Schemas |
+---------+
| foo     |
| public  |
+---------+
Total Rows: 2
Cost 8 ms
> use foo;
Total Rows: 0
Cost 4 ms
Using foo
[foo] >
```

> - Create table. The REPL supports multi-line input (just to remember to end your input in semicolon to make it a valid SQL):

Reviewer: ```suggestion
- To create a table. REPL supports multi-line input (remember to end your input with a semicolon to make it a valid SQL statement):
```

```text
[foo] > create table t(
  x STRING, 
  ts TIMESTAMP TIME INDEX);
Affected Rows: 0
Cost 12 ms
```

> - Insert some data and select them:

Reviewer: ```suggestion
- To insert or select data:
```
